sa/precertificates.go

```diff
@@ -95,12 +95,15 @@ func (ssa *SQLStorageAuthority) AddPrecertificate(ctx context.Context, req *sapb
 			args = append(args, req.IssuerID)
 		}
 
-		_, err = ssa.dbMap.WithContext(ctx).Exec(fmt.Sprintf(
+		_, err = txWithCtx.Exec(fmt.Sprintf(
 			"INSERT INTO certificateStatus (%s) VALUES (%s)",
 			csFields,
 			strings.Join(qmarks, ","),
 		), args...)
 		if err != nil {
+			if db.IsDuplicate(err) {
```

Comment:
It seems like the `ssa.dbMap` -> `txWithCtx` change is enough to fix the bug at hand. Making certificateStatus inserts return with a `berrors.DuplicateError` is in keeping with our general practice, but I think may introduce unintended consequences here. For instance, if there is somehow a similar inconsistency in the future, this could cause the CA to believe an orphan is fully integrated when only the certificateStatus entry exists. The CA would remove the orphan from its queue and forget about it.

I think to be conservative, we should omit this `berrors.DuplicateError` change for now, unless it's crucial to the fix in a way that I'm missing.